R/dataframe.R

```diff
@@ -182,24 +182,63 @@ as_data_frame.matrix <- function(x, ...) {
   as_data_frame(as.data.frame(x))
 }
 
-#' Convert row names to an explicit variable.
+#' Conversion between rownames and a column in data frame
+#'
+#' @details add_rownames: Convert row names to an explicit variable.
 #'
 #' @param df Input data frame with rownames.
 #' @param var Name of variable to use
 #' @export
+#' @rdname rownames
+#' @return add_rownames: c("tbl_df", "tbl", "data.frame")
 #' @examples
 #' tbl_df(mtcars)
 #'
-#' add_rownames(mtcars)
+#' (mtcars_tbl <- add_rownames(mtcars) )
 add_rownames <- function(df, var = "rowname") {
   stopifnot(is.data.frame(df))
 
+  if (var %in% colnames(df) ) {
+    stop(paste("There is a column named", var, "already!"))
+  }
+
   rn <- as_data_frame(setNames(list(rownames(df)), var))
   rownames(df) <- NULL
 
   as_data_frame(cbind(rn, df))
+
+}
+
+
+#' @details use_as_rownames: Inverted operation of add_rownames, returning a data.frame
+#'
+#' @rdname rownames
+#' @return use_as_rownames: data.frame
+#' @export
+#' @examples
+#'
+#' head ( use_as_rownames(mtcars_tbl) )
+#'
+use_as_rownames <- function(df, var = "rowname") {
```

Comment:
As you suggest, keep `add_rownames` function. For `use_as_rownames`, probably change it as `column_to_rownames` now. But I have not get your idea of the 2nd and 3rd paragraph. What is `class "as is"`? so far, `use_as_rownames` returns data.frame. What's the difference between `tbl_df(rownames_to_column(x))` and  `rownames_to_column(tbl_df(x)`?